yarn-project/archiver/src/archiver/archiver.test.ts

```diff
@@ -95,28 +97,29 @@ function makeL2BlockProcessedEvent(l1BlockNum: bigint, l2BlockNum: bigint) {
 /**
  * Makes a fake UnverifiedData event for testing purposes.
  * @param l1BlockNum - L1 block number.
- * @param l2BlockNum - L2Block number.
+ * @param l2Block - The l2Block this event is associated with.
  * @returns An UnverifiedData event log.
  */
-function makeUnverifiedDataEvent(l1BlockNum: bigint, l2BlockNum: bigint) {
+function makeUnverifiedDataEvent(l1BlockNum: bigint, l2Block: L2Block) {
   return {
     blockNumber: l1BlockNum,
     args: {
-      l2BlockNum,
-      sender: EthAddress.random().toString(),
+      l2BlockNum: BigInt(l2Block.number),
+      l2BlockHash: `0x${l2Block.getCalldataHash().toString('hex')}`,
+      sender: EthAddress.random(),
```

Comment:
```suggestion
      sender: EthAddress.random().toString(),
```
This should let you remove the nasty `as unknown` in the cast. See https://github.com/AztecProtocol/aztec-packages/pull/648.

Reply:
Nice, done.